src/EventStore.js

```diff
@@ -73,6 +78,14 @@ class EventStore extends Store {
     const res = ops.slice(startIndex).slice(0, amount)
     return res
   }
+  _procEntry(entry) {
+    var {op, value} = entry.payload;
+    switch(op) {
+      case "ADD": {
+        this.events.emit("db.append", value)
```

Comment:
I'm wondering two things:

1. Can this be extended for all the other options for `op` as well?
2. If so, can this be generalized away from a switch statement to something more generic that just reports back what each op does?

Know what I'm saying? In the case of the `eventstore` there are no other ops of course, but even the slightly-augmented `feedstore` would benefit from this.

If this is possible we may be able to roll this up to `orbit-db-store` and enable all of the substores with this capaibility, which would be cool.

Thoughts @haadcode and @shamb0t?

Reply:
1. Yes
2. So a universal function that does this for all the different DB types? I like the idea, but what if this wants to be extended for other purposes. Let's say that when some operation is received/written other operations are done aside from just emitting an event for higher level code. In the case of KeyValue store, an event could be triggered when a certain key is modified. That would take extra processing to compare the value being set to already existing value. Example `db.keymodified.${KeyName}`. Each store type could implement events differently.

Reply:
Thought about this some more and I'd still like to see a base-level, default behavior that's fairly generalized. Something like 

```JavaScript
this.events.emit(`db.${op}`, value)
```

The practice for the other stores is to inherit and override these methods anyway, so there's still nothing stopping anybody from implementing a different interpretation in another store. This could potentially even be moved _downward_ to `orbit-db-store` if the generalization works well.

See: https://github.com/orbitdb/orbit-db-store/blob/master/src/Store.js#L308, which handles a lot of the progress checking and event handler adding/removing for you as well

Reply:
I agree having a set of base-level events is important. I think its best to have oplog specific events that can be built upon, and guaranteed to be there. That could be implemented in `orbit-db-store`. And a separate set of events for database/store *specific* operations. 
```javascript
this.events.emit(`oplog.${op}`, value);
this.events.emit(`db.append`, value); // Eventlog append
```
Overall it will easier to build new store types, and utilize orbit-db by relying on these new events.

Reply:
I like this approach. Let's start here. 
1. PR against `orbit-db-store` with the oplog events
2. This PR with `orbit-db-eventstore` 

Sounds good?

Reply:
@aphelionz That sounds good.